ansible/playbook.yaml

```diff
@@ -0,0 +1,105 @@
+---
+- name: System Preparation
+  hosts: managed
+  remote_user: ec2-user
+  become: yes
+  tasks:
+    - name: Install Git
+      yum:
+        name: git
+        state: present
+
+    - name: Enable Amazon Linux Extras repository for Python 3.8
+      command: amazon-linux-extras enable python3.8
+      args:
+        creates: /etc/yum.repos.d/amzn2extra-python3.8.repo
+
+    - name: Install Python 3.8
+      yum:
+        name: python38
+        state: present
+
+    - name: Create /opt/todoapp directory
+      file:
+        path: /opt/todoapp
+        state: directory
+        owner: ec2-user
+        group: ec2-user
+        mode: '0755'
+
+- name: Software Installation and Setup
+  hosts: managed
+  remote_user: ec2-user
+  tasks:
+    - name: Install Poetry via curl
+      shell: curl -sSL https://install.python-poetry.org | python3.8 -
+      environment:
+        POETRY_VERSION: 1.4.1
+      args:
+        creates: ~/.local/bin/poetry
+
+    - name: Clone To Do app repository
+      become: yes
+      become_user: ec2-user
+      git:
+        repo: 'https://github.com/tkachikoti/DevOps-Course-Starter.git'
+        dest: '/opt/todoapp'
+        version: 'main'
+
+    - name: Set Python 3.8 for Poetry
+      shell: "~/.local/bin/poetry env use python3.8"
+      args:
+        chdir: /opt/todoapp
+
+    - name: Install To Do app dependencies using Poetry
+      become: yes
+      become_user: ec2-user
+      command:
+        cmd: '~/.local/bin/poetry install'
+        chdir: '/opt/todoapp'
+
+- name: To-Do App Configuration
+  hosts: managed
+  remote_user: ec2-user
+  become: yes
```

Comment:
I'd avoid putting `become` at the play level, even if it's necessary for all tasks, because that might change in future - if someone adds a new task that doesn't need root access then it's helpful if they don't automatically use it